[W13-3] ConTAct

hannahrawr opened the pull request:
A student management system for Teaching Assistants.

@alyip98 
@cyrusguo 
@preethi-d
@yogtew

parvathy-sudhir-pillai reviewed (commented):
The changes you made to the User and Developer guides are not seen as netlify still triggers the preview with  AB4's pages.
You need to change the URL in the attribute repoURL in DeveloperGuide.adoc and UserGuide.adoc with the URL of your fork. 
Please change the link and commit again.

parvathy-sudhir-pillai commented on docs/UserGuide.adoc:
```diff
@@ -94,6 +94,66 @@ Edits the phone number and email address of the 1st person to be `91234567` and
+Examples:
+
+*`sort NAME` +
+Returns a list of students by their name
```
Returns a list of students 'sorted' by their name

parvathy-sudhir-pillai commented on docs/UserGuide.adoc:
```diff
@@ -94,6 +94,66 @@ Edits the phone number and email address of the 1st person to be `91234567` and
+Examples:
+
+*`attd CS2103-W13` +
+Outputs the list of students in the class, and waits for the user to input the attendance accordingly.
```
No command to input the attendance? 
Keep in mind that all inputs to the app are to be made from CLI.

parvathy-sudhir-pillai commented on docs/UserGuide.adoc:
```diff
@@ -94,6 +94,66 @@ Edits the phone number and email address of the 1st person to be `91234567` and
+*`sort NAME` +
+Returns a list of students by their name
+*`sort GRADES` +
+Returns a list of students by their grades
```
How do you capture Grade? It is not a field in Person?

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,33 +834,155 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+Use case ends.
+
+[none]
+* 2a. The list is empty
```
ConTAct does not respond if the list is empty?

parvathy-sudhir-pillai commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,33 +834,155 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
++
+Use case ends.
+
+*Extesions*
```
Spellcheck

parvathy-sudhir-pillai reviewed (commented):
Overall: Good job, team on meeting all requirements for the v1.1 team component. However, some areas need more work. Hope you perform better in those areas in the upcoming versions.

* Readme.adoc
1. All links/references to AB-4's pages/names should be removed.
![screen shot 2018-10-07 at 4 25 51 am](https://user-images.githubusercontent.com/9066342/46575627-419b6280-c9eb-11e8-867d-8f5250b2817f.png)
2. The second badge for AppVeyor is still pointing to AB-4 repo. Has to be changed/removed.
3. The top navigation bar, LO's should go off eventually.
4. AB-4 should be acknowledged in Acknowledgements
5. View On GitHub still links to AB-4 repo

* About Us.adoc
1. The role and responsibilities are interchanged? @andrewtanJS & @cyrusguo 
2. Responsibilities for @yogtew should follow the same style as others (next line)

* Contact Us.adoc
1. Has to be updated to yours.
2. Email to be changed

* User Guide.adoc
1. Update QuickStart instructions (still refers to addressbook.jar)
2. Update FAQ
3. Features don't seem to be linked properly. You add a person to your ConTAct. How do you know/capture which group the person belongs to?
4. How (command) do you mark attendance after listing out students using attd command?
5. Where do you record student Grade to be able to sort by grade?

* Developer Guide.adoc
1. Use Case "Extesions"? Spellcheck

parvathy-sudhir-pillai reviewed (commented):
### Feedback for v1.2

**User Guide**
* Many parts of your product features, for e.g. commands, still refer to addressbook. Change these to your product's name. Further, since your product caters to TAs, you may want to change descriptions such as _Adding_ a _person_, to _Adding a student_. Same applies to all further commands.
* attd CS2103-W13 only outputs the list of students in the class. Is there a command to input the attendance?
* Emailing students has email INDEX as the command in the Product Features, but email GROUP_NAME [MORE_GROUP_NAMES] as the format in the Command Summary. Be consistent, so that you don't confuse the user (your peer who reviews your product). 

**Developer guide**
* You may remove parts irrelevant to your product. 
* Start changing your Setting Up instructions to refer to how to run your product and not addressbook
* Use sequence diagrams to explain clearly your feature implementations and how the interactions between sub-components within Logic, Model, Storage are executed. Take a look at the examples:
https://se-edu.github.io/addressbook-level4/images/DeletePersonSdForLogic.png
https://se-edu.github.io/addressbook-level4/images/UndoRedoSequenceDiagram.png 
But keep in mind that, we discourage low level diagrams that span multiple components (e.g. Logic and Model). Try to draw diagrams at multiple levels: one that shows component level interactions and others shown low level interactions within one component.
* In design considerations, the descriptions of alternatives still refer to addressbook. Is it intentional?
* Use case add students repeated

### Overall
* Good effort, Team! However, catch up on documentation work. 
* Not enough new/updated UML diagrams to give feedback on. Please try to add more soon. Include detailed UML diagrams (sequence and class) for your new features. Keep in mind that we discourage low level diagrams that span multiple components. Try to draw diagrams at multiple levels: one that shows component level interactions and others shown low level interactions within one components.
* Let's push the product towards a user-testable state by v1.3

parvathy-sudhir-pillai reviewed (commented):
### Mid v1.4 Review

* Update the link on GitHub to your project's webpage
![weblink](https://user-images.githubusercontent.com/9066342/48134418-d63bfd80-e2d4-11e8-8ff2-73f99bb8c8d2.png)
* The link in About Us page  https://CS2103-AY1819S1-W13-3 does not work. Also, set up your PPP pages ASAP.
* Attendance command in User Guide example and description use different commands
**Attendance : attendance INDEX_NUMBER at/ATTENDANCE e.g. attd CS2103-Tutorial-W13**
* Update sequence diagrams for each feature to have multiple levels. First level should include high level interactions between components. Second level diagram should show within-component interactions. Don't show interactions between components in the second level. For e.g. update the email feature diagram which shows low-level interactions between Logic and Model.
* Maintain the multi-level style sequence diagrams for other features also. Include behaviour diagrams (activity or sequence) for Mark feature as well.

**All the best for your final presentation and demo!**

cyrusguo committed:
modify existing main UI Window for compatibility with GraphPanel

cyrusguo committed:
Add GraphPage for displaying of Graph

cyrusguo committed:
Add Javascript, CSS, and JSON files as graph depedency

cyrusguo committed:
Add temporary placeholder for reference

preethi-d committed:
Update UserGuide.adoc

preethi-d committed:
Update preethi-d.adoc

cyrusguo committed:
fix travis bugs

yogtew committed:
Add tests to increase coverage

preethi-d committed:
add Email feature functional, not fully clean

preethi-d committed:
Merge remote-tracking branch 'origin/master'

yogtew committed:
Remove unused method

preethi-d committed:
Merge pull request #128 from yogtew/docs

Update Project Docs

preethi-d committed:
Merge pull request #131 from yogtew/v1.4_attendance

Attendance Command uses Mark

yogtew committed:
Merge pull request #12 from CS2103-AY1819S1-W13-3/master

update fork

yogtew committed:
Remove sort command from user guide

yogtew committed:
Remove sort from command summary

yogtew committed:
Merge pull request #133 from preethi-d/master

add Email feature functional

yogtew committed:
Merge pull request #13 from CS2103-AY1819S1-W13-3/master

Update fork

preethi-d committed:
Merge pull request #16 from CS2103-AY1819S1-W13-3/master

updating fork

hannahrawr committed:
Merge pull request #135 from yogtew/docs

Remove Sort Command from User Guide

preethi-d committed:
Changed colour scheme of app

hannahrawr committed:
Add more tests for ConTAct

These tests are largely focused around the Calendar feature. Notable additions include the ScheduleCommandSystemTest and other UI tests.

Some bug fixes were also made as a result of these tests.

hannahrawr committed:
Merge pull request #136 from preethi-d/master

Change colour scheme of application

alyip98 committed:
Merge branch 'master' into mark-tests

alyip98 committed:
Mark unit tests, tons of bug fixes

alyip98 committed:
UI updates for Mark

Mark views are persistent through edits

alyip98 committed:
MarkCommandParser tests

alyip98 committed:
Style fix

yogtew committed:
Merge pull request #14 from CS2103-AY1819S1-W13-3/master

update fork for new ui

yogtew committed:
Merge pull request #15 from yogtew/master

update branch

alyip98 committed:
Bug fix

alyip98 committed:
style fix

yogtew committed:
Code cleanup for Attendance class and AttendanceCommand class

yogtew committed:
Delete address_book_32.png

yogtew committed:
Added new icon

cyrusguo committed:
fix coveralls

preethi-d committed:
Merge pull request #17 from CS2103-AY1819S1-W13-3/master

updating

yogtew committed:
Code cleanup for AttendanceCommand Test, AttendanceCommandParser and AttendanceCommandParserTest

cyrusguo committed:
improve coveralls

yogtew committed:
Add attendance tag

yogtew committed:
Update PPP

yogtew committed:
Update User guide and Dev guide section in PPP

yogtew committed:
Revert last commit changes

preethi-d committed:
Update DeveloperGuide.adoc

yogtew committed:
Add end attendance tag

yogtew committed:
Minor edit to user guide

yogtew committed:
Update Dev guide

yogtew committed:
Edit minor inconsiste

cyrusguo committed:
add graph panel feature and tests

cyrusguo committed:
Merge pull request #1 from cyrusguo/graph-branch

add graph panel feature and tests

hannahrawr committed:
Merge pull request #139 from yogtew/docs

Update App icon

yogtew committed:
Minor edits to PPP

hannahrawr committed:
Merge pull request #140 from yogtew/v1.4-final

Clean up code

hannahrawr committed:
Merge pull request #141 from yogtew/v1.4-final-docs

Edits to Docs

yogtew committed:
Merge pull request #137 from andrewtanJS/event-tests

Add more tests for ConTAct

yogtew committed:
Merge pull request #127 from andrewtanJS/v1.4-docs

Add PPP for andrewtanJS

yogtew committed:
Minor edit to PPP

hannahrawr committed:
Merge pull request #143 from yogtew/v1.4-final-docs

Update yogtew PPP

yogtew committed:
Merge pull request #144 from preethi-d/master

Add end tag to developer guide's email description

preethi-d committed:
Update DeveloperGuide.adoc

cyrusguo committed:
fix travis and coverall

cyrusguo committed:
Merge pull request #2 from cyrusguo/graph-branch

fix travis and coverall

cyrusguo committed:
remove old data files

cyrusguo committed:
fix checkstyle

cyrusguo committed:
Merge branch 'master' of github.com:cyrusguo/main

cyrusguo committed:
fix graph panel test

cyrusguo committed:
fix stylin

alyip98 committed:
Removed DebugModelManager

cyrusguo committed:
add test for graph command

cyrusguo committed:
fix whitespaces

alyip98 committed:
Revert ModelManager visibility modifiers

alyip98 committed:
Added UML

alyip98 committed:
Merge pull request #138 from alyip98/mark-tests

Mark unit tests, minor GUI changes, various bug fixes and stability improvements

alyip98 committed:
Fixed a bug where undo and redo will leave behind expired marks

alyip98 committed:
Halfway through

alyip98 committed:
Great refactor part 2

alyip98 committed:
Added an alias for GroupCommand

alyip98 committed:
Merge branch 'master' into mark-to-group-refactor

alyip98 committed:
Refactor part 3

alyip98 committed:
Merge branch 'mark-to-group-refactor' of https://github.com/alyip98/ConTAct-alfred into mark-to-group-refactor

preethi-d committed:
Merge pull request #18 from CS2103-AY1819S1-W13-3/master

updating fork

preethi-d committed:
Finalize Userguide Email

yogtew committed:
Remove sort command use case

yogtew committed:
Update MSS for attendance

preethi-d committed:
Email Sequence Diagram

yogtew committed:
Remove Product Survey appendix

yogtew committed:
Minor update for yogtew PPP

yogtew committed:
Remove trailing whitespace

preethi-d committed:
Request Sequence Diagram

preethi-d committed:
Delete Request Sequence Diagram.png

preethi-d committed:
Request Sequence Diagram

preethi-d committed:
Updated UML diagrams

alyip98 committed:
Merge branch 'master' into dev

alyip98 committed:
Populate sampleData

yogtew committed:
Merge pull request #148 from alyip98/mark-to-group-refactor

Mark to Group refactor

yogtew committed:
Merge pull request #16 from CS2103-AY1819S1-W13-3/master

Update fork

alyip98 committed:
Update alfred's PPP

alyip98 committed:
Merge branch 'master' into dev

preethi-d committed:
Update DeveloperGuide.adoc

preethi-d committed:
Update DeveloperGuide.adoc

preethi-d committed:
Delete Request Sequence Diagram.png

preethi-d committed:
Update Sequence Diagram

preethi-d committed:
Merge pull request #149 from yogtew/v1.4-final-docs

Update Dev guide

preethi-d committed:
Spacings

preethi-d committed:
Update DeveloperGuide.adoc

yogtew committed:
Merge pull request #18 from CS2103-AY1819S1-W13-3/master

update v1.4-final-docs

preethi-d committed:
Updated use case

preethi-d committed:
Add manual testing for email command

alyip98 committed:
Group documentation changes

preethi-d committed:
Correct manual testing for email

alyip98 committed:
Manual tests section updates

preethi-d committed:
Component Interactions Email

preethi-d committed:
Update DeveloperGuide.adoc

preethi-d committed:
Clean email feature explanation and add flow UML

alyip98 committed:
More documentation updates

alyip98 committed:
Bug fix

preethi-d committed:
Update UserGuide.adoc

preethi-d committed:
Update UserGuide.adoc

yogtew committed:
Merge pull request #151 from preethi-d/master

Added UMLs and manual tests and edited DG and UG

preethi-d committed:
Removed 2nd occurence of email

preethi-d committed:
Update UserGuide.adoc

yogtew committed:
Add manual tests for attendance

yogtew committed:
Update PPP for yogtew

yogtew committed:
Minor edit to attendance command message usage

yogtew committed:
Minor edits to UG

yogtew committed:
Edit UG to include given suggestions

preethi-d committed:
Merge pull request #152 from yogtew/v1.4-final-docs

Minor doc updates

cyrusguo committed:
Merge remote-tracking branch 'upstream/master' into merge-panel

cyrusguo committed:
fix panel display issues and coveralls

hannahrawr committed:
Merge pull request #153 from cyrusguo/merge-panel

Add Graph Feature

yogtew committed:
Merge pull request #19 from CS2103-AY1819S1-W13-3/master

Test

cyrusguo committed:
change graph message

yogtew committed:
Merge pull request #154 from cyrusguo/master

change graph message

hannahrawr committed:
Add sample data and fix minor bugs

hannahrawr committed:
Add EoF

preethi-d committed:
Merge pull request #155 from andrewtanJS/update-sampledata

Add sample data and fix minor bugs

hannahrawr committed:
Merge branch 'master' into dev

alyip98 committed:
Update src/test/java/seedu/address/logic/parser/GroupCommandParserTest.java

alyip98 committed:
Merge branch 'dev' of https://github.com/alyip98/ConTAct-alfred into dev

preethi-d committed:
Editing PPP

alyip98 committed:
Update docs/UserGuide.adoc

cyrusguo committed:
Merge pull request #3 from cyrusguo/merge-panel

Merge panel

preethi-d committed:
Adding tags

cyrusguo committed:
Merge branch 'master' of github.com:cyrusguo/main

cyrusguo committed:
add docs for graph

cyrusguo committed:
add image

hannahrawr committed:
Update UML for model

preethi-d committed:
Adding proposed additions for 2.0

preethi-d committed:
Adding contributions to dev guide

preethi-d committed:
Update preethi-d.adoc

preethi-d committed:
Update preethi-d.adoc

cyrusguo committed:
add SD and CI file

alyip98 committed:
last min bug fixes hoho

hannahrawr committed:
Update UML architecture diagrams

yogtew committed:
Delete johndoe.adoc

cyrusguo committed:
add devguide for graph

cyrusguo committed:
Merge branch 'master' of github.com:cyrusguo/main

cyrusguo committed:
add graph mockup

cyrusguo committed:
Merge branch 'master' of github.com:cyrusguo/main

yogtew committed:
Fix spelling

cyrusguo committed:
update graph doc

alyip98 committed:
more last min bug fixes

cyrusguo committed:
update graphCI image

hannahrawr committed:
Fix minor errors referencing person instead of student

yogtew committed:
Merge pull request #157 from preethi-d/master

Edit PPP and DG

cyrusguo committed:
fix image issues

yogtew committed:
Fix spelling errors

preethi-d committed:
Manual test header

preethi-d committed:
Merge pull request #159 from preethi-d/master

Manual test header

cyrusguo committed:
add manual testing

yogtew committed:
Update DeveloperGuide.adoc

preethi-d committed:
Delete Component Interactions Email.png

preethi-d committed:
Component Interactions Email

preethi-d committed:
Delete Component Interactions Email.png

preethi-d committed:
Component Interactions Email

preethi-d committed:
Delete Component Interactions Email.png

preethi-d committed:
Component Interactions Email

yogtew committed:
Update PPP

alyip98 committed:
Fixed GUI bug

alyip98 committed:
Update docs/DeveloperGuide.adoc

yogtew committed:
Merge pull request #161 from preethi-d/master

Edit uml

preethi-d committed:
Merge pull request #158 from yogtew/v1.4-final-docs

Update Docs

alyip98 committed:
Merge branch 'master' into dev

preethi-d committed:
Update DeveloperGuide.adoc

preethi-d committed:
Merge branch 'master' into master

alyip98 committed:
Merge branch 'master' into dev

preethi-d committed:
Update preethi-d.adoc

preethi-d committed:
Update DeveloperGuide.adoc

yogtew committed:
Update DeveloperGuide.adoc

yogtew committed:
Update yogtew.adoc

yogtew committed:
Update DeveloperGuide.adoc

preethi-d committed:
Update DeveloperGuide.adoc

yogtew committed:
Merge branch 'master' into v1.4-final-docs

preethi-d committed:
Update DeveloperGuide.adoc

yogtew committed:
Update DeveloperGuide.adoc

preethi-d committed:
Update DeveloperGuide.adoc

preethi-d committed:
Merge pull request #150 from alyip98/dev

Populated SampleData with 20 students, updated PPP

hannahrawr committed:
update images and manual testing section

yogtew committed:
Merge branch 'master' into v1.4-final-docs

yogtew committed:
Merge pull request #20 from CS2103-AY1819S1-W13-3/master

Test fork

hannahrawr committed:
Fix style issue

hannahrawr committed:
Merge branch 'master' into v1.4-docs-andrew

preethi-d committed:
Merge pull request #162 from preethi-d/master

Edit tag

preethi-d committed:
Merge pull request #163 from yogtew/v1.4-final-docs

Minor Doc edits

preethi-d committed:
Merge pull request #156 from andrewtanJS/v1.4-docs-andrew

Updating UML diagrams and documentation

hannahrawr committed:
Merge branch 'master' into master

yogtew committed:
Delete Ui.png

yogtew committed:
Add new UI mockup

yogtew committed:
Delete Ui.png

yogtew committed:
Add new UI Mockup

yogtew committed:
Update README.adoc

hannahrawr committed:
Merge pull request #160 from cyrusguo/master

Update Graph Docs

yogtew committed:
Update yogtew.adoc

preethi-d committed:
Update DeveloperGuide.adoc

preethi-d committed:
Update preethi-d.adoc

yogtew committed:
Merge branch 'master' into test

preethi-d committed:
Merge pull request #165 from preethi-d/master

Update PPP

preethi-d committed:
Merge pull request #164 from yogtew/test

Update UI Mockup

cyrusguo committed:
add Cyrus PPP

cyrusguo committed:
add Data encryption

cyrusguo committed:
edit devguide

cyrusguo committed:
edit ppp doc

preethi-d committed:
Update DeveloperGuide.adoc

yogtew committed:
Merge pull request #166 from cyrusguo/master

add Cyrus PPP

preethi-d committed:
Merge pull request #167 from preethi-d/master

Update PPP and DG

hannahrawr committed:
Update PPP for Andrew'

hannahrawr committed:
Merge branch 'v1.4-docs-andrew' of https://github.com/andrewtanJS/addressbook-level4 into v1.4-docs-andrew

yogtew committed:
Update DeveloperGuide.adoc

preethi-d committed:
Merge pull request #168 from andrewtanJS/v1.4-docs-andrew

Added PPP for Andrew

hannahrawr committed:
Add project portfolio links

hannahrawr committed:
Merge pull request #169 from yogtew/test

Update Developer Guide

preethi-d committed:
Update DeveloperGuide.adoc

hannahrawr committed:
Fix reference

hannahrawr committed:
Fix graph reference

preethi-d committed:
Delete Sequence Diagram LOGIC.png

preethi-d committed:
equence Diagram Logic edited

cyrusguo committed:
Merge branch 'master' of github.com:cyrusguo/main

preethi-d committed:
Merge pull request #19 from CS2103-AY1819S1-W13-3/master

Updating fork

hannahrawr committed:
Fix documentation

hannahrawr committed:
Fix image

preethi-d committed:
Update UserGuide.adoc

hannahrawr committed:
Merge pull request #172 from andrewtanJS/master

Fix documentation

hannahrawr committed:
Merge pull request #170 from cyrusguo/master

update GraphCI image

preethi-d committed:
Remove encryption part

yogtew committed:
Update yogtew.adoc

hannahrawr committed:
Merge pull request #171 from preethi-d/master

Final touch ups

preethi-d committed:
UML edit

preethi-d committed:
Update DeveloperGuide.adoc

hannahrawr committed:
Update UserGuide.adoc

yogtew committed:
Add Ui.png
